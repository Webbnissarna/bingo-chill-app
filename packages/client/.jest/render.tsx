import React from "react";
import { render as rtlRender } from "@testing-library/react";

export function render(ui: React.ReactElement) {
  return rtlRender(<>{ui}</>);
}
