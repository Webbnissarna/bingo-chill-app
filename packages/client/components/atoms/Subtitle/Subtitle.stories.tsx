import type { Story, StoryDefault } from "@ladle/react";
import Subtitle from ".";

interface StoryProps {
  text: string;
}

function StoryComponent({ text }: StoryProps) {
  return <Subtitle>{text}</Subtitle>;
}

export const Default: Story<StoryProps> = StoryComponent;

Default.args = {
  text: "Hello World",
};

Default.argTypes = {
  text: {
    control: { type: "text" },
  },
};

export default {
  title: "Atoms / Subtitle",
} satisfies StoryDefault;
