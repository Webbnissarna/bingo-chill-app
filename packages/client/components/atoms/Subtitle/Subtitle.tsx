interface SubtitleProps {
  children: string;
}

export default function Subtitle({ children }: SubtitleProps): JSX.Element {
  return <h1 className="font-heading text-snowStorm-2 text-3xl">{children}</h1>;
}
