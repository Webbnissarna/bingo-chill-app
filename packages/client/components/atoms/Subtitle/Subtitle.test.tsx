import { render } from "../../../.jest/render";
import Subtitle from ".";

describe("Subtitle", () => {
  it("renders", () => {
    const { asFragment } = render(<Subtitle>Hello World</Subtitle>);

    expect(asFragment()).toMatchSnapshot();
  });
});
