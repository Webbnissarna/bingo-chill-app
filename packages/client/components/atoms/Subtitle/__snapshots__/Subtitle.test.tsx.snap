// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subtitle renders 1`] = `
<DocumentFragment>
  <h1
    class="font-heading text-snowStorm-2 text-3xl"
  >
    Hello World
  </h1>
</DocumentFragment>
`;
